tests/director/external-services-mock-integration/async_api_delete_test.go

```diff
@@ -0,0 +1,171 @@
+			},
+		},
+	}
+
+	t.Log("Get Dex token")
+	dexToken, err := idtokenprovider.GetDexToken()
+	require.NoError(t, err)
+	dexGraphQLClient := gql.NewAuthorizedGraphQLClient(dexToken)
+
+	t.Log(fmt.Sprintf("Registering application: %s", appName))
+	appInputGQL, err := tc.Graphqlizer.ApplicationRegisterInputToGQL(appInput)
+	require.NoError(t, err)
+	registerRequest := fixRegisterApplicationRequest(appInputGQL)
+	app := graphql.ApplicationExt{}
+	err = tc.RunOperationWithCustomTenant(ctx, dexGraphQLClient, testConfig.DefaultTenant, registerRequest, &app)
+	require.NoError(t, err)
+
+	defer deleteApplicationOnExit(t, ctx, dexGraphQLClient, app.ID, testConfig.DefaultTenant)
+
+	t.Log("Unlock the mock application webhook")
+	unlockWebhook(t, operationFullPath)
+	isInDesiredState, webhookResponse := isWebhookOperationInDesiredState(t, operationFullPath, webhook.OperationResponseStatusOK)
+	require.True(t, isInDesiredState, fmt.Sprintf("Webhook responded with: %s. Expected state: %s", webhookResponse, webhook.OperationResponseStatusOK))
+
+	t.Log("Start async Delete of application")
+	unregisterAsyncApplicationInTenant(t, ctx, dexGraphQLClient, app.ID, testConfig.DefaultTenant)
+
+	t.Log("Prepare operation client for compass-system namespace")
+	cfg, err := rest.InClusterConfig()
+	require.NoError(t, err)
+	k8sClient, err := client.NewForConfig(cfg)
+	require.NoError(t, err)
+	operationsK8sClient := k8sClient.Operations("compass-system")
+
+	operationName := fmt.Sprintf("application-%s", app.ID)
+	t.Log(fmt.Sprintf("Check operation CR with name %s is created", operationName))
+	operation, err := operationsK8sClient.Get(ctx, operationName, metav1.GetOptions{})
+	require.NoError(t, err)
+	require.NotEmpty(t, operation)
+
+	t.Log(fmt.Sprintf("Verify operation CR with name %s is in progress", operationName))
+	counter := 0
+	isInDesiredState, webhookResponse = isWebhookOperationInDesiredState(t, operationFullPath, webhook.OperationResponseStatusINProgress)
+	for (!isInDesiredState) && counter < 100 {
+		t.Log(fmt.Sprintf("[%s] Webhook responded with: %s. Expected state: %s", time.Now().Format("06.01.02 15:04:05"), webhookResponse, webhook.OperationResponseStatusINProgress))
+		time.Sleep(time.Second * 5)
+		isInDesiredState, webhookResponse = isWebhookOperationInDesiredState(t, operationFullPath, webhook.OperationResponseStatusINProgress)
+		counter++
+	}
+	isInDesiredState, webhookResponse = isWebhookOperationInDesiredState(t, operationFullPath, webhook.OperationResponseStatusINProgress)
+	require.True(t, isInDesiredState, fmt.Sprintf("Webhook responded with: %s. Expected state: %s", webhookResponse, webhook.OperationResponseStatusINProgress))
+
+	t.Log("Verify the application status in director is 'ready:false'")
+	application := getApplication(t, ctx, dexGraphQLClient, testConfig.DefaultTenant, app.ID)
+	require.False(t, application.Ready, "Application is not in status ready:false")
```

Comment:
Tweak the fixtures and also assert the `application.Error` field that it is empty. Additionally assert the timestamps, most notably the deletion timestamp that it's not null (and it's around the current time? because you cannot assert an exact time)